tests/unittests/server_view_tests.py

```
@@ -1303,3 +1312,337 @@ def test_filter_valid_id(self):
+        stages = {stage.__name__: stage for stage in fosg._STAGES}
+        for definition in fosd.stage_definitions():
+            signature = inspect.signature(stages[definition.name].__init__)
+            for param in definition.params:
+                has_default = (
+                    signature.parameters[param.name].default
+                    is not inspect.Parameter.empty
+                )
+                self.assertEqual(
+                    param.required,
+                    not has_default,
+                    "%s.%s" % (definition.name, param.name),
```

Ruff (0.15.21)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)